en/03_Drawing_a_triangle/00_Setup/03_Physical_devices_and_queue_families.adoc

```diff
@@ -22,108 +25,90 @@ void pickPhysicalDevice() {
 }
 ----
 
-The graphics card that we'll end up selecting will be stored in a VkPhysicalDevice handle that is added as a new class member.
-This object will be implicitly destroyed when the VkInstance is destroyed, so we won't need to do anything new in the `cleanup` function.
+The graphics card that we'll end up selecting will be stored in a
+VkPhysicalDevice handle added as a new class member.
 
 [,c++]
 ----
-VkPhysicalDevice physicalDevice = VK_NULL_HANDLE;
+vk::raii::PhysicalDevice physicalDevice;
 ----
 
-Listing the graphics cards is very similar to listing extensions and starts with querying just the number.
+Listing the graphics cards is very similar to listing extensions and starts with
+querying just the number.
 
 [,c++]
 ----
-uint32_t deviceCount = 0;
-vkEnumeratePhysicalDevices(instance, &deviceCount, nullptr);
+auto devices = instance.enumeratePhysicalDevices()
 ----
 
-If there are 0 devices with Vulkan support then there is no point going further.
+If there are no devices with Vulkan support, then there is no point going
+further.
 
 [,c++]
 ----
-if (deviceCount == 0) {
+if (devices.empty()) {
     throw std::runtime_error("failed to find GPUs with Vulkan support!");
 }
 ----
 
-Otherwise we can now allocate an array to hold all of the VkPhysicalDevice handles.
-
-[,c++]
-----
-std::vector<VkPhysicalDevice> devices(deviceCount);
-vkEnumeratePhysicalDevices(instance, &deviceCount, devices.data());
-----
-
-Now we need to evaluate each of them and check if they are suitable for the operations we want to perform, because not all graphics cards are created equal.
-For that we'll introduce a new function:
-
-[,c++]
-----
-bool isDeviceSuitable(VkPhysicalDevice device) {
-    return true;
-}
-----
-
-And we'll check if any of the physical devices meet the requirements that we'll add to that function.
+Now we need to evaluate each of them and check if they are suitable for the
+operations we want to perform, because not all graphics cards are created equal.
+We'll check if any of the physical devices meet the requirements that we'll
+add to that function.
 
 [,c++]
 ----
 for (const auto& device : devices) {
-    if (isDeviceSuitable(device)) {
-        physicalDevice = device;
-        break;
-    }
-}
-
-if (physicalDevice == VK_NULL_HANDLE) {
-    throw std::runtime_error("failed to find a suitable GPU!");
+    physicalDevice = std::make_unique<vk::raii::PhysicalDevice>(device);
+    break;
 }
 ----
 
-The next section will introduce the first requirements that we'll check for in the `isDeviceSuitable` function.
-As we'll start using more Vulkan features in the later chapters we will also extend this function to include more checks.
-
 == Base device suitability checks
```

Comment:
We should consider rewriting or removing this chapter all along and instead replace it with explicit device selection. This "device suitability check" has caused lots of confusion judging from community feedback.

Reply:
This might be good for after the merge to revisit.  I agree, this needs to be revisited.